### PR TITLE
fix: Removetime_range_endpoints from query context object pt 2

### PR DESCRIPTION
**Pull request by hughhhh**
### SUMMARY

original reference this time we actually save the data properly 😪
https://github.com/apache/superset/pull/19423

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/migrations/versions/cecc6bf46990_rm_time_range_endpoints_2.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        query_context = json.loads(slc.query_context)
+    except json.decoder.JSONDecodeError:
+        pass
+
+    queries = query_context.get("queries")
+
+    for query in queries:
+        query.get("extras", {}).pop("time_range_endpoints", None)
+
+    slc.query_context = json.dumps(query_context)
```
Comment (hughhhh) [inline superset/migrations/versions/cecc6bf46990_rm_time_range_endpoints_2.py:58]: initial we were saving it back to `slc.queries` and that doesn't exist on the Slice model

File: superset/migrations/versions/cecc6bf46990_rm_time_range_endpoints_2.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+def upgrade():
+    bind = op.get_bind()
+    session = db.Session(bind=bind)
+    for slc in session.query(Slice).filter(
+        Slice.query_context.like("%time_range_endpoints%")
+    ):
+        upgrade_slice(slc)
+
+    session.commit()
```
Comment (eschutho) [inline superset/migrations/versions/cecc6bf46990_rm_time_range_endpoints_2.py:69]: do we log anything if this doesn't work?

File: superset/migrations/versions/cecc6bf46990_rm_time_range_endpoints_2.py (hunk 1 above)
Comment (eschutho) [reply superset/migrations/versions/cecc6bf46990_rm_time_range_endpoints_2.py:58]: good catch!

File: tests/integration_tests/migrations/cecc6bf46990_rm_time_range_endpoints_2__tests.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+def test_upgrade():
+    slc = Slice(slice_name="FOO", query_context=json.dumps(sample_query_context))
+
+    upgrade_slice(slc)
+
+    query_context = json.loads(slc.query_context)
+    queries = query_context.get("queries")
+    for q in queries:
+        extras = q.get("extras", {})
+        assert "time_range_endpoints" not in extras
```
Comment (eschutho) [inline tests/integration_tests/migrations/cecc6bf46990_rm_time_range_endpoints_2__tests.py:92]: nice! A test on a migration!

File: tests/integration_tests/migrations/cecc6bf46990_rm_time_range_endpoints_2__tests.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+            "custom_form_data": {},
+            "post_processing": [],
+        }
+    ],
+    "form_data": {
+        "viz_type": "table",
+        "datasource": "27__table",
+        "slice_id": 545,
+        "url_params": {},
+        "time_range_endpoints": ["inclusive", "exclusive"],
```
Comment (eschutho) [inline tests/integration_tests/migrations/cecc6bf46990_rm_time_range_endpoints_2__tests.py:56]: total nit, but technically this wouldn't be on this payload.

**Review by eschutho — APPROVED** [review]

**Comment by ktmud** [comment]
maybe not necessary, but id the original migration doesn't work, maybe we can change it to noop?

File: superset/migrations/versions/cecc6bf46990_rm_time_range_endpoints_2.py [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+    id = sa.Column(sa.Integer, primary_key=True)
+    query_context = sa.Column(sa.Text)
+    slice_name = sa.Column(sa.String(250))
+
+
+def upgrade_slice(slc: Slice):
+    try:
+        query_context = json.loads(slc.query_context)
+    except json.decoder.JSONDecodeError:
+        pass
```
Comment (craig-rueda) [inline superset/migrations/versions/cecc6bf46990_rm_time_range_endpoints_2.py:51]: You should return here, as the json parse failed

File: tests/integration_tests/migrations/cecc6bf46990_rm_time_range_endpoints_2__tests.py [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+        "force": False,
+        "result_format": "json",
+        "result_type": "full",
+    },
+    "result_format": "json",
+    "result_type": "full",
+}
+
+
+def test_upgrade():
```
Comment (craig-rueda) [inline tests/integration_tests/migrations/cecc6bf46990_rm_time_range_endpoints_2__tests.py:83]: Can you add a test for the case when the query_context is non-json?

File: superset/migrations/versions/cecc6bf46990_rm_time_range_endpoints_2.py (hunk 2 above)
Comment (hughhhh) [reply superset/migrations/versions/cecc6bf46990_rm_time_range_endpoints_2.py:69]: it should throw if i'm not mistaken

**Review by craig-rueda — APPROVED** [review]

**Review by eschutho — APPROVED** [review]

**Comment by sadpandajoe** [comment]
🏷️  preset:2022.15

**Pull request merged**